src/color.rs

```diff
@@ -31,6 +31,18 @@ pub enum ColorType {
 
 }
 
+impl ColorType {
+    /// Returns the number of bits contained in a pixel of this `ColorType`
+    pub fn bits_per_pixel(&self) -> usize {
```

Comment:
Return type `usize` makes a limited amount of sense. It's the one of the internal method, true, but its internal use is also restricted to: 

* casts to `u32`
* comparison to constants `0`, `1`, `2`, `3`
* as a buffer index (which may be the reason for the internal choice)

As a public method however we may prefer a platform independent type, for which `u8` may be too small for the future (think four 64-bit components) but `u16` should be plenty. Also, `u16` can be converted into `usize` using `Into` since `1.26.1` so this is would not be a loss of functionality for internal use.

@fintelia Any comments, also with regards to the in-progress `ColorType` rework?

Reply:
I agree that there should be a public method exposing this information. The current plan with the `ColorType` rework is to have a `ColorType::bytes_per_pixel` function and make all pixels an integer number of bytes -- which is good for everyone's sanity.

Reply:
For `bytes_per_pixel`, which data type would provide enough range?

Reply:
A `u8` should be fine, right? I don't know of any pixel format with more than 4 channels, but this could fit up to 31 x `f64` channels.

Reply:
Should I change the internal free functions too, or only change the public interface?

Reply:
It would be cleaner to change the internal functions as well. All current use should be able to be replaced with proper `_.into()` or `u32::from(_)` conversion I think.